app/helpers/herbaria_helper.rb

```diff
@@ -2,12 +2,10 @@
 
 # helper methods for Herbaria views
 module HerbariaHelper
-  def herbarium_top_users(herbarium)
-    Herbarium.connection.select_rows(%(
-      SELECT u.name, u.login, COUNT(u.id)
-      FROM herbarium_records hr JOIN users u ON u.id = hr.user_id
-      WHERE hr.herbarium_id = #{herbarium.id}
-      GROUP BY u.id ORDER BY COUNT(u.id) DESC LIMIT 5
-    ))
+  def herbarium_top_users(herbarium_id)
+    User.joins(:herbarium_records).
+      where(HerbariumRecord[:herbarium_id].eq(herbarium_id)).
+      select(User[:name], User[:login], User[:id].count).
+      group(User[:id]).order(User[:id].count.desc).take(5)
```

Comment:
If you felt like it, you could eliminate some of the Arel.
But it may read better with all Arel.
```ruby
    User.joins(:herbarium_records).
      where(herbarium_records: { herbarium_id: herbarium_id }).
      select(:name, :login, User[:id].count).
      group(:id).order(User[:id].count.desc).take(5)
```

Reply:
I like it. Committed it, thanks!